src/app/utils/field-validation/validators/__tests__/email-validation.spec.ts

```diff
@@ -295,7 +295,35 @@ describe('Email field validation', () => {
     expect(validateResult._unsafeUnwrap()).toEqual(true)
   })
 
-  it('should allow any valid email address when isVerifiable is false and hasAllowedEmailDomains is true, regardless of the cardinality of allowedEmailDomains', () => {
+  it('should allow any email address with a domain in allowedEmailDomains when isVerifiable is true and hasAllowedEmailDomains is false, and allowedEmailDomains is not empty', () => {
```

Comment:
Not too familiar with the feature/codebase yet. Shouldn't hasAllowedEmailDomains be true if allowedEmailDomains is not empty?

Reply:
Thanks! This is a good point. I will add a model-layer check on this (and accompanying test). But I think this test should remain as representation of the email validator itself

Reply:
[6fdf9d0](https://github.com/opengovsg/FormSG/pull/6497/commits/6fdf9d0154475032f75f30d9426520ecddf38914)